### PR TITLE
Add CodeMirror update and usage of requireJS to migration guide

### LINKED ISSUE #186: Add CodeMirror update and usage of requireJS to migration guide
See WoltLab/WCF#4277

### PR DESCRIPTION
**Pull request by ghost**
Close #186

### CODE REVIEW

File: docs/migration/wsc54/javascript.md [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# Migrating from WSC 5.4 - TypeScript and JavaScript
+
+## CodeMirror
+
+CodeMirror, which we use for highlighting custom CSS, for example, has been updated to version 5.61.1 and we now also deliver all supported languages/modes.
```
Comment (TimWolla) [inline docs/migration/wsc54/javascript.md]: highlighting is not fitting.

File: docs/migration/wsc54/javascript.md [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+To properly support all languages/modes, CodeMirror is now loaded via requireJS, which requires the original structure of the CodeMirror package, i.e. `codemirror.js` being in a `lib` folder.
```
Comment (TimWolla) [inline docs/migration/wsc54/javascript.md]:
```suggestion
To properly support all languages/modes, CodeMirror is now loaded via the AMD module loader, which requires the original structure of the CodeMirror package, i.e. `codemirror.js` being in a `lib` folder.
```

(Technically it's not RequireJS, but Alameda)

File: docs/migration/wsc54/javascript.md [hunk 3]
```diff
@@ -0,0 +1,10 @@
...
+To preserve backward-compatibility, we also keep copies of `codemirror.js` and `codemirror.css` in version 5.61.1 directly in `js/3rdParty/codemirror`.
+These files are, however, considered deprecated and you should migrate to using requireJS (see `codemirror` ACP template).
```
Comment (TimWolla) [inline docs/migration/wsc54/javascript.md]:
```suggestion
These files are, however, considered deprecated and you should migrate to using `require()` (see `codemirror` ACP template).
```

File: docs/migration/wsc54/javascript.md [hunk 4]
```diff
@@ -0,0 +1,10 @@
...
+
+See [WCF#4277](https://github.com/WoltLab/WCF/pull/4277) for more information.
```
Comment (TimWolla) [inline docs/migration/wsc54/javascript.md]:
```suggestion
See [WoltLab/WCF#4277](https://github.com/WoltLab/WCF/pull/4277) for more information.
```

File: docs/migration/wsc54/javascript.md (hunk 1 above)
Comment (ghost) [inline docs/migration/wsc54/javascript.md]:
```suggestion
CodeMirror, the code editor we use for editing templates and SCSS, for example, has been updated to version 5.61.1 and we now also deliver all supported languages/modes.
```

**Pull request merged**